src/components/CodeView/index.spec.tsx

```diff
@@ -1,19 +1,54 @@
 import * as React from 'react';
-import { shallow } from 'enzyme';
+import { Link } from 'react-router-dom';
+import { Location } from 'history';
+import { mount } from 'enzyme';
 
 import refractor from '../../refractor';
 import { mapWithDepth } from './utils';
 import { getLanguageFromMimeType } from '../../utils';
+import {
+  createContextWithFakeRouter,
+  createFakeLocation,
+  shallowUntilTarget,
+} from '../../test-helpers';
 import styles from './styles.module.scss';
 
-import CodeView from '.';
+import CodeView, { CodeViewBase, PublicProps, scrollToSelectedLine } from '.';
 
 describe(__filename, () => {
+  type RenderParams = Partial<PublicProps> & {
+    location?: Location<{}>;
+  };
+
   const render = ({
-    content = 'some content',
-    mimeType = 'mime/type',
-  } = {}) => {
-    return shallow(<CodeView mimeType={mimeType} content={content} />);
+    location = createFakeLocation(),
+    ...otherProps
+  }: RenderParams = {}) => {
+    const props = {
+      content: 'some content',
+      mimeType: 'mime/type',
+      ...otherProps,
+    };
+
+    return shallowUntilTarget(<CodeView {...props} />, CodeViewBase, {
+      shallowOptions: createContextWithFakeRouter({ location }),
+    });
+  };
+
+  const renderWithMount = ({
+    location = createFakeLocation(),
+    ...otherProps
+  }: RenderParams = {}) => {
+    const props = {
```

Comment:
This can be factored into a shared function like `getRenderProps()`

Reply:
right, thanks!